webpack/InsightsCloudSync/Components/InsightsSettings/InsightsSettingsActions.js

```diff
@@ -0,0 +1,60 @@
+import API from 'foremanReact/API';
+import { insightsCloudUrl } from '../../InsightsCloudSyncHelpers';
+import {
+  INSIGHTS_SYNC_SETTING_SET,
+  INSIGHTS_SYNC_SETTING_SET_ERROR,
+  INSIGHTS_SYNC_SETTINGS_GET_SUCCESS,
+  INSIGHTS_SYNC_SETTINGS_GET_FAILURE,
+} from './InsightsSettingsConstants';
+
+export const getInsightsSyncSettings = () => async dispatch => {
+  try {
+    const {
+      data: { insightsSyncEnabled },
+    } = await API.get(insightsCloudUrl('settings'));
+    dispatch({
+      type: INSIGHTS_SYNC_SETTINGS_GET_SUCCESS,
+      payload: {
+        settings: {
+          insightsSyncEnabled,
+        },
+      },
+    });
+  } catch (error) {
+    dispatch({
+      type: INSIGHTS_SYNC_SETTINGS_GET_FAILURE,
+      payload: {
+        settings: {
+          error: error.message,
+        },
+      },
+    });
+  }
+};
+
+export const setInsightsSyncEnabled = currentInsightsSyncEnabled => async dispatch => {
+  try {
+    const {
+      data: { insightsSyncEnabled },
+    } = await API.patch(insightsCloudUrl('settings'), {
```

Comment:
Interesting, I looked on the settings page, and saw they are using `PUT` there for setting update,
after reading a bit more about REST here: https://blog.fullstacktraining.com/restful-api-design-post-vs-put-vs-patch/

it feels that `PUT` might be a better option here, since it's replacing the setting's value and clearing what was before
though it's not a deal breaker for me :)

Reply:
Since the API call is at the `settings` level but the call updates only one setting, I think the proper verb would be patch.
Although Rails doesn't distinguish between the two.